CONTRIBUTING.md

```diff
@@ -0,0 +1,83 @@
+# Textkernel OneUI Developer Guide
+
+When contributing to the OneUI library, please consider the following developer guidelines.
+
+## Definition of Done Checklist
+Any new implementation is expected to meet the following standards before it can be merged.
+
+- The component has been manually tested and complies with the Textkernel [browser support guidelines](https://textkernel.com/browser-support/).
+- The component has a [__displayName__](#display-names) defined.
+- The component comes with a [__detailed propTypes__](#component-props) (and defaultProps) definition.
+- The implementation complies with [__accessibility__](#accessibility) standards.
+- The component is __properly exported__ in `src/index.js` (maintain alphabetical order).
+- The implementation is thoroughly [__covered by tests__](#testing) (Jest / Enzyme, preferably 100%), not only with snapshots.
+- The component [__comes with a story__](#component-showcases) in Storybook that allows props to be changed with knobs.
+- The implementation is __reviewed__ by another contributor.
+- The complete __build is passing__ (including tests and code linting).
+
+## Component Design
+In general: OneUI tries to adhere to the principles prescribed by [Atomic Design](http://bradfrost.com/blog/post/atomic-web-design/). This means we intend to work from the smallest possible units (molecules) towards larger compounds (organisms).
+
+### Functional vs. Class Components
+Simple functional components have the preference over class / stateful components. Only use class components if it needs to have internal methods or a state of its own.
```

Comment:
Only use class components if it needs to have ~~internal~~ **lifecycle** methods or a state of its own.